### ISSUE TITLE
don't pollute /mingw{32,64}/lib with Perl modules

### ISSUE THREAD

**Issue by nponeccop** [Alexpux/MINGW-packages#58]
[ArchLinux build](https://projects.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/perl) uses `-D` options to `Configure` to override default locations for `core`, `vendor` and `site` library and binary paths.

As all Windows systems are (almost) the same, and running `configure` on Windows is a tough task, Perl tries to avoid it. All variables normally set by `configure` are hardcoded in `win32/config.gc`. `gc` stands for Gnu Compiler I think.

I think Msys2 should follow ArchLinux, with `/usr/` basically replaced by `.../mingw{32,64}/`. The following should be converted into a patch for `win32/config.gc`:

```
    -Dprefix=/usr -Dvendorprefix=/usr \
    -Dprivlib=/usr/share/perl5/core_perl \
    -Darchlib=/usr/lib/perl5/core_perl \
    -Dsitelib=/usr/share/perl5/site_perl \
    -Dsitearch=/usr/lib/perl5/site_perl \
    -Dvendorlib=/usr/share/perl5/vendor_perl \
    -Dvendorarch=/usr/lib/perl5/vendor_perl \
    -Dscriptdir=/usr/bin/core_perl \
    -Dsitescript=/usr/bin/site_perl \
    -Dvendorscript=/usr/bin/vendor_perl \
```

It may be also useful to take a look at the build process used by Strawberry perl, arguably the best perl distribution for mingw.

It's quite cumbersome, but here are patches it applies:

http://api.metacpan.org/source/KMX/Perl-Dist-Strawberry-4.002/share/perl-5.20/

And here is its build procedure for core perl:

https://metacpan.org/source/KMX/Perl-Dist-Strawberry-4.002/lib/Perl/Dist/Strawberry/Step/InstallPerlCore.pm

It still invokes `dmake` at line 151, so I don't see a reason to change our bulid process, but we may take some variables it sets.

**Comment by elieux** [comment]
I think @renatosilva solved this a while ago. Can we close this, @nponeccop?

**Comment by nponeccop** [comment]
Close. I haven't used it for a long time and I'm on a slow connection now
so I can't easily check. I'll reopen it if I discover problems later on.

**Comment by renatosilva** [comment]
This is fixed, see fb6ca8a.
